app-modules/assist-data-model/src/Filament/Resources/StudentResource/RelationManagers/EngagementsRelationManager.php

```diff
@@ -0,0 +1,127 @@
+                        TextEntry::make('description'),
+                    ]),
+
+                RepeatableEntry::make('deliverables')
+                    ->columnSpanFull()
+                    ->schema([
+                        TextEntry::make('channel'),
+                        IconEntry::make('delivery_status')
+                            ->icon(fn (EngagementDeliveryStatus $state): string => match ($state) {
+                                EngagementDeliveryStatus::SUCCESSFUL => 'heroicon-o-check-circle',
+                                EngagementDeliveryStatus::AWAITING => 'heroicon-o-clock',
+                                EngagementDeliveryStatus::FAILED => 'heroicon-o-x-circle',
+                            })
+                            ->color(fn (EngagementDeliveryStatus $state): string => match ($state) {
+                                EngagementDeliveryStatus::SUCCESSFUL => 'success',
+                                EngagementDeliveryStatus::AWAITING => 'info',
+                                EngagementDeliveryStatus::FAILED => 'danger',
+                            }),
+                        TextEntry::make('delivered_at'),
+                        TextEntry::make('delivery_response'),
+                    ])
+                    ->columns(2),
+            ]);
+    }
+
+    public function table(Table $table): Table
+    {
+        return $table
+            ->recordTitleAttribute('id')
+            ->columns([
+                TextColumn::make('id'),
+                TextColumn::make('subject'),
+                TextColumn::make('description'),
+                TextColumn::make('channels')
+                    ->label('Delivery Channels')
+                    ->state(function (Engagement $record) {
+                        return $record->deliverables->pluck('channel')->map(function ($channel) {
+                            return $channel->name;
+                        })->implode(', ');
+                    }),
+            ])
+            ->filters([
+            ])
+            ->headerActions([
+                CreateAction::make()
+                    ->after(function (Engagement $engagement, array $data) {
+                        $this->afterCreate($engagement, $data['delivery_methods']);
+                    }),
+            ])
+            ->actions([
+                ViewAction::make(),
+            ])
+            ->bulkActions([
+            ])
+            ->emptyStateActions([
+                CreateAction::make()
+                    ->after(function (Engagement $engagement, array $data) {
+                        $this->afterCreate($engagement, $data['delivery_methods']);
+                    }),
+            ]);
+    }
+
+    public function afterCreate(Engagement $engagement, array $deliveryMethods): void
+    {
+        $createDeliverablesForEngagement = resolve(CreateDeliverablesForEngagement::class);
+
+        $createDeliverablesForEngagement($engagement, $deliveryMethods);
+    }
```

Comment:
Just curious about your thoughts on whether or not this is something that should be handled by `created` event/listeners for `Engagement`?

Reply:
I think it certainly can. Either way, we need to hook into functionality after the model is created in Filament, because we are reliant on the extra form fields that provide this information. But that's probably a slightly better way to handle this.